lib/lintSource.js

```diff
@@ -88,7 +87,7 @@ function lintPostcssResult(
   postcssResult.stylelint.quiet = config.quiet
 
   const newlineMatch = postcssResult.root.toResult().css.match(/\r?\n/)
-  const newline = newlineMatch ? newlineMatch[0] : os.EOL
+  const newline = newlineMatch ? newlineMatch[0] : "\n"
```

Comment:
Failing tests: https://ci.appveyor.com/project/stylelint/stylelint/build/4183. Visually diffs the same, but actually they have different line endings.

In tests we might have no line endings in sources. Rule `at-rule-empty-line-before: "always"`. Test case:

```js
reject: [{
	code: "a {} @media {}",
	fixed: "a {}\n\n @media {}",
	message: messages.expected,
}]
```

On Unix-like system test passes, but on Window it doesn't. Because it expects `a {}\r\n\r\n @media {}`, because there were no line ending in the source, so it uses `os.EOL` as fallback.

We need to find a way to have tests cross-platform.